allennlp/semparse/domain_languages/wikitables_language.py

```diff
@@ -3,75 +3,23 @@
 # Unfortunately, mypy doesn't like this very much, so we have to "type: ignore" a bunch of things.
 # But it makes for a nicer induced grammar, so it's worth it.
 from numbers import Number
-from typing import Dict, List, NamedTuple, Set, Tuple
+from typing import Dict, List, NamedTuple, Set, Tuple, Any
 import logging
 import re
 
 from allennlp.semparse.domain_languages.domain_language import (DomainLanguage, ExecutionError,
                                                                 PredicateType, predicate)
 from allennlp.semparse.contexts.table_question_knowledge_graph import MONTH_NUMBERS
 from allennlp.semparse.contexts import TableQuestionContext
+from allennlp.semparse.contexts.table_question_context import Date, CellValueType
```

Comment:
I'm not sure how you're getting away with this.  It looks like a circular import to me.  I guess because the other import goes directly to `domain_language.py`?  Either way, this should be fixed.  `CellValueType` is easy enough to just redefine here, instead of having the import.

Well, actually, now that I look at this a bit more, maybe it's right for the imports to go this way, instead of the other way, as I thought...  Still, the `Date` class and its tests should be somewhere more general.  Any suggestions?

Reply:
I like your suggestion to move `Date` into `allennlp.semparse.domain_languages.common.date`. I'm doing that.